internal/clients/api_client.go

```diff
@@ -503,6 +542,92 @@ func buildAlertingClient(baseConfig BaseConfig, config kibana.Config) *alerting.
 	return alerting.NewAPIClient(&alertingConfig)
 }
 
+func buildFleetClient(d *schema.ResourceData, baseConfig BaseConfig) (*fleet.Client, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	// Order of precedence for config options:
+	// 1 (highest): environment variables
+	// 2: resource config
+	// 3: base config
+
+	// Set variables from base config.
+	config := fleet.Config{
+		Username: baseConfig.Username,
+		Password: baseConfig.Password,
+	}
+
+	// Set variables from resource config.
+	fleetDataRaw, ok := d.GetOk("fleet")
+	if !ok {
+		return nil, diags
+	}
+	fleetData := fleetDataRaw.([]interface{})[0].(map[string]any)
+	if !ok {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to parse Fleet configuration",
+			Detail:   "Fleet configuration data has not been configured correctly or is empty",
+		})
+		return nil, diags
+	}
+	if v, ok := fleetData["endpoint"].(string); ok && v != "" {
+		config.URL = v
+	}
+	if v, ok := fleetData["username"].(string); ok && v != "" {
+		config.Username = v
+	}
+	if v, ok := fleetData["password"].(string); ok && v != "" {
+		config.Password = v
+	}
+	if v, ok := fleetData["api_key"].(string); ok && v != "" {
+		config.APIKey = v
+	}
+	if v, ok := fleetData["ca_certs"].([]interface{}); ok && len(v) > 0 {
+		for _, elem := range v {
+			if vStr, elemOk := elem.(string); elemOk {
+				config.CACerts = append(config.CACerts, vStr)
+			}
+		}
+	}
+	if v, ok := fleetData["insecure"].(bool); ok {
+		config.Insecure = v
+	}
+
+	// Set variables from environment variables.
+	if v := os.Getenv("FLEET_ENDPOINT"); v != "" {
```

Comment:
I was told by the Fleet API team that Fleet may not always be under Kibana, so I separated it out here. I could change some of these back to KIBANA_*, but there are some values that won't carry over, since there isn't an option for it in the config (API_KEY for example). Schema would probably have to change too, so it would get a bit messy.